test-framework/core/src/main/java/org/keycloak/testframework/crypto/CryptoKeyStore.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.keycloak.testframework.crypto;
+
+import org.junit.jupiter.api.Assumptions;
+import org.keycloak.common.crypto.CryptoIntegration;
+import org.keycloak.common.util.CertificateUtils;
+import org.keycloak.common.util.KeyUtils;
+import org.keycloak.common.util.KeystoreUtil;
+import org.keycloak.common.util.PemUtils;
+import org.keycloak.common.util.Time;
+import org.keycloak.representations.idm.CertificateRepresentation;
+
+import javax.crypto.SecretKey;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class CryptoKeyStore {
```

Comment:
there is already this util: [GenerateKeystoreForTestUtil](https://github.com/keycloak/keycloak/blob/main/tests/utils/src/main/java/org/keycloak/tests/utils/admin/GenerateKeystoreForTestUtil.java)

don't you want to remove it and refactor also this test: https://github.com/keycloak/keycloak/blob/main/tests/base/src/test/java/org/keycloak/tests/admin/client/CredentialsTest.java?

Reply:
Nopes, we can do that as a follow-up. This PR is already big enough as is